opensearch-operator/pkg/reconcilers/scaler.go

```diff
@@ -33,14 +35,17 @@ func NewScalerReconciler(
 	recorder record.EventRecorder,
 	reconcilerContext *ReconcilerContext,
 	instance *opsterv1.OpenSearchCluster,
-	opts ...reconciler.ResourceReconcilerOption,
+	opts ...ReconcilerOption,
 ) *ScalerReconciler {
```

Comment:
May I know what is change doing by adding `ReconcilerOption` ?

Reply:
That change is needed to, for one, make this reconciler behave more like the others, and also to get the osClientTransport field from the ReconcilerOptions which is a needed argument for the `CreateClientForCluster` function. In the end this also helps with testing as the RoundTripper can be mocked.